Add new aws_workspaces_pool resource

JaydonSunley opened the pull request:
This adds new resource & datasource `aws_workspaces_pool`.

### Relations

Closes #38823

### References

- https://docs.aws.amazon.com/workspaces/latest/api/API_WorkspacesPool.html
- https://docs.aws.amazon.com/workspaces/latest/api/API_CreateWorkspacesPool.html


### Output from Acceptance Testing

```console
% make testacc PKG=workspaces TESTS='TestAccWorkSpacesDataSource_serial/Pool|TestAccWorkSpaces_serial/Pool'
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.8 test ./internal/service/workspaces/... -v -count 1 -parallel 20 -run='TestAccWorkSpacesDataSource_serial/Pool|TestAccWorkSpaces_serial/Pool'  -timeout 360m -vet=off
2025/05/20 22:03:29 Initializing Terraform AWS Provider...
=== RUN   TestAccWorkSpacesDataSource_serial
=== PAUSE TestAccWorkSpacesDataSource_serial
=== RUN   TestAccWorkSpaces_serial
=== PAUSE TestAccWorkSpaces_serial
=== CONT  TestAccWorkSpacesDataSource_serial
=== CONT  TestAccWorkSpaces_serial
=== RUN   TestAccWorkSpacesDataSource_serial/Pool
=== RUN   TestAccWorkSpaces_serial/Pool
=== RUN   TestAccWorkSpacesDataSource_serial/Pool/basic
=== RUN   TestAccWorkSpaces_serial/Pool/basic
=== PAUSE TestAccWorkSpacesDataSource_serial/Pool/basic
=== CONT  TestAccWorkSpacesDataSource_serial/Pool/basic
=== PAUSE TestAccWorkSpaces_serial/Pool/basic
=== RUN   TestAccWorkSpaces_serial/Pool/disappears
=== PAUSE TestAccWorkSpaces_serial/Pool/disappears
=== RUN   TestAccWorkSpaces_serial/Pool/ApplicationSettings
=== PAUSE TestAccWorkSpaces_serial/Pool/ApplicationSettings
=== RUN   TestAccWorkSpaces_serial/Pool/TimeoutSettings
=== PAUSE TestAccWorkSpaces_serial/Pool/TimeoutSettings
=== RUN   TestAccWorkSpaces_serial/Pool/TimeoutSettings_maxUserDurationInSeconds
=== PAUSE TestAccWorkSpaces_serial/Pool/TimeoutSettings_maxUserDurationInSeconds
=== CONT  TestAccWorkSpaces_serial/Pool/basic
=== CONT  TestAccWorkSpaces_serial/Pool/TimeoutSettings
=== CONT  TestAccWorkSpaces_serial/Pool/ApplicationSettings
=== CONT  TestAccWorkSpaces_serial/Pool/disappears
=== CONT  TestAccWorkSpaces_serial/Pool/TimeoutSettings_maxUserDurationInSeconds
--- PASS: TestAccWorkSpacesDataSource_serial (126.13s)
    --- PASS: TestAccWorkSpacesDataSource_serial/Pool (0.00s)
        --- PASS: TestAccWorkSpacesDataSource_serial/Pool/basic (126.13s)
--- PASS: TestAccWorkSpaces_serial (166.02s)
    --- PASS: TestAccWorkSpaces_serial/Pool (0.00s)
        --- PASS: TestAccWorkSpaces_serial/Pool/TimeoutSettings_maxUserDurationInSeconds (120.77s)
        --- PASS: TestAccWorkSpaces_serial/Pool/basic (127.46s)
        --- PASS: TestAccWorkSpaces_serial/Pool/TimeoutSettings (145.19s)
        --- PASS: TestAccWorkSpaces_serial/Pool/ApplicationSettings (149.19s)
        --- PASS: TestAccWorkSpaces_serial/Pool/disappears (166.02s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/workspaces	171.397s
...
```

JaydonSunley committed:
Add aws_workspaces_pool resource

JaydonSunley committed:
Add pool tests
Refactor Pool resource

JaydonSunley committed:
Add pools resource documentation

JaydonSunley committed:
Cleanup tests
Organise schema
Add findPoolByName for datasource
Add missing value to flattenApplicationSettings
Update docs

JaydonSunley committed:
Add WorkSpaces Pools datasource

JaydonSunley committed:
Linting

JaydonSunley committed:
Add changelog

JaydonSunley committed:
Update documentation for aws_workspaces_pool data source argument reference

JaydonSunley committed:
Fix semgrep error for ARN check

JaydonSunley committed:
convert workspaces_pool resource to use plugin framework instead of SDK

hanoj-budime commented:
Please respond — we’re waiting for the IaC support feature, and no one has replied yet. Kindly merge this PR as soon as possible.

@justinretzolk 
@jar-b 
@ewbankkit
@johnsonaj
@JaydonSunley

4iter4life commented:
hm.. nobody need it.. lets make a workarroung via null_resource and aws cli?

subham-ibmhc committed:
cleanup schema and tests

subham-ibmhc committed:
f/aws_workspaces_pool: Cleanup schema, fix tests, documentation and changelog, move sweeper

```console
% make t T=TestAccWorkSpaces_serial/Pool/ K=workspaces
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-workspaces-pool 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/workspaces/... -v -count 1 -parallel 20 -run='TestAccWorkSpaces_serial/Pool/'  -timeout 360m -vet=off
2026/07/01 17:15:24 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/01 17:15:24 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccWorkSpaces_serial
=== PAUSE TestAccWorkSpaces_serial
=== CONT  TestAccWorkSpaces_serial
=== RUN   TestAccWorkSpaces_serial/Pool
=== RUN   TestAccWorkSpaces_serial/Pool/basic
=== PAUSE TestAccWorkSpaces_serial/Pool/basic
=== RUN   TestAccWorkSpaces_serial/Pool/disappears
=== PAUSE TestAccWorkSpaces_serial/Pool/disappears
=== RUN   TestAccWorkSpaces_serial/Pool/applicationSettings
=== PAUSE TestAccWorkSpaces_serial/Pool/applicationSettings
=== RUN   TestAccWorkSpaces_serial/Pool/timeoutSettings
=== PAUSE TestAccWorkSpaces_serial/Pool/timeoutSettings
=== CONT  TestAccWorkSpaces_serial/Pool/basic
=== CONT  TestAccWorkSpaces_serial/Pool/applicationSettings
=== CONT  TestAccWorkSpaces_serial/Pool/timeoutSettings
=== CONT  TestAccWorkSpaces_serial/Pool/disappears
--- PASS: TestAccWorkSpaces_serial (137.29s)
    --- PASS: TestAccWorkSpaces_serial/Pool (0.00s)
        --- PASS: TestAccWorkSpaces_serial/Pool/disappears (114.29s)
        --- PASS: TestAccWorkSpaces_serial/Pool/basic (121.74s)
        --- PASS: TestAccWorkSpaces_serial/Pool/applicationSettings (124.39s)
        --- PASS: TestAccWorkSpaces_serial/Pool/timeoutSettings (137.29s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/workspaces 142.736s
```

subham-ibmhc committed:
f/aws_workspaces_pool: Add RI

```console
% make t T=TestAccWorkSpaces_serial/Pool/Identity K=workspaces
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-workspaces-pool 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/workspaces/... -v -count 1 -parallel 20 -run='TestAccWorkSpaces_serial/Pool/Identity'  -timeout 360m -vet=off
2026/07/02 08:18:02 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/02 08:18:02 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccWorkSpaces_serial
=== PAUSE TestAccWorkSpaces_serial
=== CONT  TestAccWorkSpaces_serial
=== RUN   TestAccWorkSpaces_serial/Pool
=== RUN   TestAccWorkSpaces_serial/Pool/Identity
=== RUN   TestAccWorkSpaces_serial/Pool/Identity/basic
=== RUN   TestAccWorkSpaces_serial/Pool/Identity/RegionOverride
--- PASS: TestAccWorkSpaces_serial (175.80s)
    --- PASS: TestAccWorkSpaces_serial/Pool (175.80s)
        --- PASS: TestAccWorkSpaces_serial/Pool/Identity (175.80s)
            --- PASS: TestAccWorkSpaces_serial/Pool/Identity/basic (90.34s)
            --- PASS: TestAccWorkSpaces_serial/Pool/Identity/RegionOverride (85.46s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/workspaces 181.138s
```

subham-ibmhc committed:
f/aws_workspaces_pool: Add list support

```console
% make t T=TestAccWorkSpacesPool_List_ K=workspaces
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-workspaces-pool 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/workspaces/... -v -count 1 -parallel 20 -run='TestAccWorkSpacesPool_List_'  -timeout 360m -vet=off
2026/07/02 09:19:47 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/02 09:19:47 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccWorkSpacesPool_List_basic
=== PAUSE TestAccWorkSpacesPool_List_basic
=== RUN   TestAccWorkSpacesPool_List_includeResource
=== PAUSE TestAccWorkSpacesPool_List_includeResource
=== RUN   TestAccWorkSpacesPool_List_regionOverride
=== PAUSE TestAccWorkSpacesPool_List_regionOverride
=== CONT  TestAccWorkSpacesPool_List_basic
=== CONT  TestAccWorkSpacesPool_List_regionOverride
=== CONT  TestAccWorkSpacesPool_List_includeResource
--- PASS: TestAccWorkSpacesPool_List_regionOverride (69.91s)
--- PASS: TestAccWorkSpacesPool_List_includeResource (82.74s)
--- PASS: TestAccWorkSpacesPool_List_basic (83.73s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/workspaces 88.637s
```

subham-ibmhc committed:
update changelog

subham-ibmhc commented:
Acceptance test results -

```console
% make t T=TestAccWorkSpaces_serial/Pool/ K=workspaces
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-workspaces-pool 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/workspaces/... -v -count 1 -parallel 20 -run='TestAccWorkSpaces_serial/Pool/'  -timeout 360m -vet=off
2026/07/01 17:15:24 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/01 17:15:24 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccWorkSpaces_serial
=== PAUSE TestAccWorkSpaces_serial
=== CONT  TestAccWorkSpaces_serial
=== RUN   TestAccWorkSpaces_serial/Pool
=== RUN   TestAccWorkSpaces_serial/Pool/basic
=== PAUSE TestAccWorkSpaces_serial/Pool/basic
=== RUN   TestAccWorkSpaces_serial/Pool/disappears
=== PAUSE TestAccWorkSpaces_serial/Pool/disappears
=== RUN   TestAccWorkSpaces_serial/Pool/applicationSettings
=== PAUSE TestAccWorkSpaces_serial/Pool/applicationSettings
=== RUN   TestAccWorkSpaces_serial/Pool/timeoutSettings
=== PAUSE TestAccWorkSpaces_serial/Pool/timeoutSettings
=== CONT  TestAccWorkSpaces_serial/Pool/basic
=== CONT  TestAccWorkSpaces_serial/Pool/applicationSettings
=== CONT  TestAccWorkSpaces_serial/Pool/timeoutSettings
=== CONT  TestAccWorkSpaces_serial/Pool/disappears
--- PASS: TestAccWorkSpaces_serial (137.29s)
    --- PASS: TestAccWorkSpaces_serial/Pool (0.00s)
        --- PASS: TestAccWorkSpaces_serial/Pool/disappears (114.29s)
        --- PASS: TestAccWorkSpaces_serial/Pool/basic (121.74s)
        --- PASS: TestAccWorkSpaces_serial/Pool/applicationSettings (124.39s)
        --- PASS: TestAccWorkSpaces_serial/Pool/timeoutSettings (137.29s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/workspaces 142.736s
```
```console
% make t T=TestAccWorkSpaces_serial/Pool/Identity K=workspaces
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-workspaces-pool 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/workspaces/... -v -count 1 -parallel 20 -run='TestAccWorkSpaces_serial/Pool/Identity'  -timeout 360m -vet=off
2026/07/02 08:18:02 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/02 08:18:02 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccWorkSpaces_serial
=== PAUSE TestAccWorkSpaces_serial
=== CONT  TestAccWorkSpaces_serial
=== RUN   TestAccWorkSpaces_serial/Pool
=== RUN   TestAccWorkSpaces_serial/Pool/Identity
=== RUN   TestAccWorkSpaces_serial/Pool/Identity/basic
=== RUN   TestAccWorkSpaces_serial/Pool/Identity/RegionOverride
--- PASS: TestAccWorkSpaces_serial (175.80s)
    --- PASS: TestAccWorkSpaces_serial/Pool (175.80s)
        --- PASS: TestAccWorkSpaces_serial/Pool/Identity (175.80s)
            --- PASS: TestAccWorkSpaces_serial/Pool/Identity/basic (90.34s)
            --- PASS: TestAccWorkSpaces_serial/Pool/Identity/RegionOverride (85.46s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/workspaces 181.138s
```

```console
% make t T=TestAccWorkSpacesPool_List_ K=workspaces
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f-workspaces-pool 🌿...
TF_ACC=1 go1.26.4 test ./internal/service/workspaces/... -v -count 1 -parallel 20 -run='TestAccWorkSpacesPool_List_'  -timeout 360m -vet=off
2026/07/02 09:19:47 Creating Terraform AWS Provider (SDKv2-style)...
2026/07/02 09:19:47 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccWorkSpacesPool_List_basic
=== PAUSE TestAccWorkSpacesPool_List_basic
=== RUN   TestAccWorkSpacesPool_List_includeResource
=== PAUSE TestAccWorkSpacesPool_List_includeResource
=== RUN   TestAccWorkSpacesPool_List_regionOverride
=== PAUSE TestAccWorkSpacesPool_List_regionOverride
=== CONT  TestAccWorkSpacesPool_List_basic
=== CONT  TestAccWorkSpacesPool_List_regionOverride
=== CONT  TestAccWorkSpacesPool_List_includeResource
--- PASS: TestAccWorkSpacesPool_List_regionOverride (69.91s)
--- PASS: TestAccWorkSpacesPool_List_includeResource (82.74s)
--- PASS: TestAccWorkSpacesPool_List_basic (83.73s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/workspaces 88.637s
```

subham-ibmhc committed:
fix CI failures

subham-ibmhc committed:
fix semgrep

subham-ibmhc committed:
add Workspaces to fixSomeInitialisms for listpages

subham-ibmhc committed:
update documentation

ewbankkit reviewed (approved):
LGTM :rocket:.

subham-ibmhc commented:
@JaydonSunley Thank you for your contribution. 🚀
